File: pages/api/integrations/slack.ts
import type { NextApiRequest, NextApiResponse } from 'next';

interface SlackResponse {
  integration: string;
  status: string;
  configured: boolean;
  timestamp: string;
  capabilities?: string[];
  workspace?: string;
  bot_name?: string;
  available_channels?: string[];
  mode?: string;
  message?: string;
  action?: string;
  message_id?: string;
  details?: Record<string, unknown>;
}

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse<SlackResponse>
) {
  const { method, query } = req;
  
  // Check if Slack is configured
  const isConfigured = Boolean(process.env.SLACK_TOKEN);
  
  if (method === 'GET') {
    const action = query.action as string || 'status';
    
    if (action === 'connect') {
      res.status(200).json({
        integration: 'Slack',
        status: 'connected',
        configured: isConfigured,
        timestamp: new Date().toISOString(),
        message: isConfigured 
          ? 'Slack connected successfully'
          : 'Slack connected in demo mode (configure SLACK_TOKEN for real integration)',
      });
    } else if (action === 'disconnect') {
      res.status(200).json({
        integration: 'Slack',
        status: 'disconnected',
        configured: isConfigured,
        timestamp: new Date().toISOString(),
        message: 'Slack disconnected successfully',
      });
    } else {
      // Return status
      res.status(200).json({
        integration: 'Slack',
        status: 'connected',
        configured: isConfigured,
        timestamp: new Date().toISOString(),
        capabilities: [
          'Post to channels',
          'Direct messages',
          'Rich formatting',
          'File uploads',
          'Thread replies',
        ],
        workspace: isConfigured ? 'Real Workspace' : 'Demo Workspace',
        bot_name: 'ProSprint Bot',
        available_channels: ['#general', '#ops', '#dev', '#alerts'],
        mode: isConfigured ? 'production' : 'demo',
      });
    }
  } else if (method === 'POST') {
    const data = req.body;
    const action = data.action || 'post_message';
    const channel = data.channel || '#general';
    const message = data.message || '';
    
    try {
      if (isConfigured) {
        // In production mode - would post to Slack here
        // const token = process.env.SLACK_TOKEN;
        // const response = await fetch('https://slack.com/api/chat.postMessage', {
        //   method: 'POST',
        //   headers: {
        //     'Authorization': `Bearer ${token}`,
        //     'Content-Type': 'application/json'
        //   },
        //   body: JSON.stringify({
        //     channel: channel,
        //     text: message
        //   })
        // });
        // const slackResponse = await response.json();
        
        res.status(200).json({
          integration: 'Slack',
          status: 'completed',
          configured: isConfigured,
          message: `Slack ${action} operation successful (production mode)`,
          timestamp: new Date().toISOString(),
          action,
          message_id: `slack_${Date.now()}`,
          details: {
            channel,
            message_length: message.length,
            message_text: message.length > 100 ? message.substring(0, 100) + '...' : message,
            workspace: 'Real Workspace',
            posted_by: 'ProSprint Bot',
            permalink: `https://slack.com/archives/C12345/${Date.now()}`,
            mode: 'production',
          },
        });
      } else {
        // Demo mode
        res.status(200).json({
          integration: 'Slack',
          status: 'completed',
          configured: isConfigured,
          message: `Slack ${action} operation successful (demo mode)`,
          timestamp: new Date().toISOString(),
          action,
          message_id: `slack_${Date.now()}`,
          details: {
            channel,
            message_length: message.length,
            message_text: message.length > 100 ? message.substring(0, 100) + '...' : message,
            workspace: 'Demo Workspace',
            posted_by: 'ProSprint Bot',
            permalink: `https://demo.slack.com/archives/C12345/${Date.now()}`,
            mode: 'demo',
            note: 'Configure SLACK_TOKEN for real Slack integration',
          },
        });
      }
    } catch (error) {
      res.status(500).json({
        integration: 'Slack',
        status: 'error',
        configured: isConfigured,
        message: `Slack operation failed: ${error instanceof Error ? error.message : 'Unknown error'}`,
        timestamp: new Date().toISOString(),
        action,
      });
    }
  } else if (method === 'OPTIONS') {
    res.setHeader('Access-Control-Allow-Origin', '*');
    res.setHeader('Access-Control-Allow-Methods', 'GET, POST, OPTIONS');
    res.setHeader('Access-Control-Allow-Headers', 'Content-Type');
    res.status(200).end();
  } else {
    res.status(405).json({
      integration: 'Slack',
      status: 'error',
      configured: isConfigured,
      message: 'Method not allowed',
      timestamp: new Date().toISOString(),
    });
  }
}

File: pages/api/integrations/slack.py
from http.server import BaseHTTPRequestHandler
import json
from datetime import datetime

class handler(BaseHTTPRequestHandler):
    """
    Slack Integration API endpoint.
    Handles Slack operations like posting messages, direct messages, and notifications.
    """
    
    def do_GET(self):
        # Set response headers
        self.send_response(200)
        self.send_header('Content-type', 'application/json')
        self.send_header('Access-Control-Allow-Origin', '*')
        self.end_headers()
        
        # Return Slack integration status
        response = {
            "integration": "Slack",
            "status": "connected",
            "timestamp": datetime.now().isoformat(),
            "capabilities": [
                "Post to channels",
                "Direct messages",
                "Rich formatting",
                "File uploads",
                "Thread replies"
            ],
            "workspace": "Demo Workspace",
            "bot_name": "ProSprint Bot",
            "available_channels": ["#general", "#ops", "#dev", "#alerts"]
        }
        
        self.wfile.write(json.dumps(response).encode())
        return
    
    def do_POST(self):
        # Get content length
        content_length = int(self.headers.get('Content-Length', 0))
        
        # Read and parse request body
        post_data = self.rfile.read(content_length)
        
        try:
            data = json.loads(post_data.decode('utf-8'))
        except json.JSONDecodeError:
            data = {}
        
        # Set response headers
        self.send_response(200)
        self.send_header('Content-type', 'application/json')
        self.send_header('Access-Control-Allow-Origin', '*')
        self.end_headers()
        
        # Process Slack action
        action = data.get('action', 'post_message')
        channel = data.get('channel', '#general')
        message = data.get('message', '')
        
        # Demo: Simulate Slack posting
        result = {
            "status": "completed",
            "message": f"Slack {action} operation successful",
            "timestamp": datetime.now().isoformat(),
            "action": action,
            "message_id": f"slack_{int(datetime.now().timestamp())}",
            "details": {
                "channel": channel,
                "message_length": len(message),
                "workspace": "Demo Workspace",
                "posted_by": "ProSprint Bot",
                "permalink": f"https://demo.slack.com/archives/C12345/{int(datetime.now().timestamp())}"
            }
        }
        
        self.wfile.write(json.dumps(result).encode())
        return
    
    def do_OPTIONS(self):
        # Handle CORS preflight
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, POST, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()
        return
